sdk/keyvault/keyvault-secrets/samples/challenge.ts

```diff
@@ -0,0 +1,87 @@
+import { SecretsClient } from "../src";
+import { EnvironmentCredential } from "@azure/identity";
+
+import {
+  ServiceClientCredentials,
+  TokenCredential,
+  isTokenCredential,
+  RequestPolicyFactory,
+  deserializationPolicy,
+  signingPolicy,
+  bearerTokenAuthenticationPolicy,
+  RequestOptionsBase,
+  exponentialRetryPolicy,
+  redirectPolicy,
+  systemErrorRetryPolicy,
+  generateClientRequestIdPolicy,
+  proxyPolicy,
+  throttlingRetryPolicy,
+  getDefaultProxySettings,
+  userAgentPolicy
+} from "@azure/core-http";
+
+import { RetryConstants, SDK_VERSION } from "../src/core/utils/constants";
+import { ChallengeBasedAuthenticationPolicy, challengeBasedAuthenticationPolicy } from "../src/core/challengeBasedAuthenticationPolicy";
+
+import { TokenCredentials, } from "@azure/core-http";
+import { Pipeline } from "../src/core/keyVaultBase";
+
+async function main(): Promise<void> {
+  // EnvironmentCredential expects the following three environment variables:
+  // - AZURE_TENANT_ID: The tenant ID in Azure Active Directory
+  // - AZURE_CLIENT_ID: The application (client) ID registered in the AAD tenant
+  // - AZURE_CLIENT_SECRET: The client secret for the registered application
+  const credential = new EnvironmentCredential();
+  let retryOptions: any = {};
+  let pipelineOptions: any = {};
+  const requestPolicyFactories: RequestPolicyFactory[] = [
+    proxyPolicy(getDefaultProxySettings((pipelineOptions.proxyOptions || {}).proxySettings)),
```

Comment:
I'm not sure if this is valid feedback, but here it is:
Could we export something from our core that would abstract all of these policies away? Something optional, but that would help users avoid this level of complexity.

Reply:
Possibly. Though some policies we're using will be specific to key vault.  Most of the policies do live in core-http already.